src/app/shared/utils/funds.ts

```diff
@@ -0,0 +1,21 @@
+import {Expense} from '../models/expense';
+
+export const numberWithSpace = (number: number): string => {
+  return number.toString().replace(/\B(?=(\d{3})+(?!\d))/g, " ");
+};
+
+export const convertToCurrency = (amount: number) => (amount / 100);
+
+export const convertToPennies = (amount: number) => (amount * 100);
+
+export const upsertExpense = (expenses: Expense[], expense: Expense): Expense[] => {
+  const elementIndex = expenses.findIndex(e => e.id === expense.id);
+
+  if (elementIndex > -1) {
```

Comment:
Minor, but the main idea of [findIndex()](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/findIndex?retiredLocale=uk) is to find the index of existing element or return `-1` when the searching result is negative. So you can check "whether the searching result wasn't negative": `if (elementIndex != -1)` but not `if (elementIndex > -1)`. Your solution will work correctly too, but it's like some sort of workaround from a logical point of view.